### PR TITLE
Add db migration testing tools to pytest-inmanta-extensions

### PR DESCRIPTION
**Pull request by andraskvr**
# Description

The dump_tool and PGRestore are required in the inmanta-lsm extension for testing database migrations in a similar manner to the core.

Related to inmanta/inmanta-lsm#240

# Self Check:

- [ ] ~Attached issue to pull request~
- [ ] ~Changelog entry~
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] ~End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~

# Reviewer Checklist:

- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Code is clear and sufficiently documented
- [x] Correct, in line with design

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,6 +13,7 @@ install:
 install-tests:
 	pip install -U setuptools pip
 	python3 tests_common/copy_files_from_core.py
+	isort -rc tests_common/
```
Comment (arnaudsjs) [inline Makefile]: Why was this line added here?

File: tests_common/README.md [hunk 2]
```diff
@@ -10,10 +10,11 @@ plugin.
 ```bash
 pip install -U setuptools pip
 python3 tests_common/copy_files_from_core.py
+isort -rc tests_common/
```
Comment (arnaudsjs) [inline tests_common/README.md]: Why was this line added here?

File: tests_common/copy_files_from_core.py [hunk 3]
```diff
@@ -43,16 +50,39 @@
...
 shutil.copy(utils_py, dest_dir)
 shutil.copy(server_crt_file, dest_data_dir)
 shutil.copy(server_open_key_file, dest_data_dir)
 shutil.copytree(enduser_certs_dir, os.path.join(dest_data_dir, "ca", "enduser-certs"))
+shutil.copy(db_common_py, dest_db_common_py)
+shutil.copy(dump_tool_py, dest_dump_tool_py)
+shutil.copytree(simple_project_dir, dest_simple_project_dir)
+
+# Fix import
+contents = []
```
Comment (arnaudsjs) [inline tests_common/copy_files_from_core.py]: Variable is never used.

File: Makefile (hunk 1 above)
Comment (andraskvr) [reply Makefile]: The import `from utils...` in dump_tool.py is changed to `from inmanta_tests.utils...` which is in the wrong position. Instead of finding the correct position 'by hand', I thought it would be easiest to just use isort on the copied files to make sure everything is in the right place.

File: tests_common/copy_files_from_core.py [hunk 4]
```diff
@@ -43,16 +50,38 @@
...
 shutil.copy(utils_py, dest_dir)
 shutil.copy(server_crt_file, dest_data_dir)
 shutil.copy(server_open_key_file, dest_data_dir)
 shutil.copytree(enduser_certs_dir, os.path.join(dest_data_dir, "ca", "enduser-certs"))
+shutil.copy(db_common_py, dest_db_common_py)
+shutil.copy(dump_tool_py, dest_dump_tool_py)
+shutil.copytree(simple_project_dir, dest_simple_project_dir)
+
+# Fix import
+with fileinput.input(dest_dump_tool_py, inplace=True) as f:
```
Comment (wouterdb) [inline tests_common/copy_files_from_core.py]: please use a consistent solution with the conftest.py
it now has

```
# Import the utils module differently when conftest is put into the inmanta_tests package
if __file__ and os.path.dirname(__file__).split("/")[-1] == "inmanta_tests":
    from inmanta_tests import utils  # noqa: F401
else:
    import utils
```

(I'm undecided about which solution is worst)
